compiler/rustc_hir_analysis/src/check/intrinsicck.rs

```diff
@@ -84,34 +106,13 @@ impl<'a, 'tcx> InlineAsmCtxt<'a, 'tcx> {
             ty::Adt(adt, substs) if adt.repr().simd() => {
                 let fields = &adt.non_enum_variant().fields;
                 let elem_ty = fields[FieldIdx::from_u32(0)].ty(self.tcx, substs);
+
                 match elem_ty.kind() {
-                    ty::Never | ty::Error(_) => return None,
```

Comment:
This needs to remain on the same level, there are 40 LoC or so that this early return would skip.

Reply:
No? Just below the match is the following:
```rust
let Some(asm_ty) = asm_ty else {
    let msg = &format!("cannot use value of type `{ty}` for inline assembly");
    let mut err = self.tcx.sess.struct_span_err(expr.span, msg);
    err.note(
        "only integers, floats, SIMD vectors, pointers and function pointers \
         can be used as arguments for inline assembly",
    );
    err.emit();
    return None;
};
```

Reply:
I consider the fact that the previous version of the code jumps out rather than even reaching that `else` and thus rather than emitting that error message to be significant.

Reply:
...basically, `!` is *totally incoherent* for `repr(simd)` to begin with, so we should not be emitting the message regarding "but that's not a valid inline assembly input?" We should just be throwing up our hands and allowing the rest of rustc to emit a more general "what are you **doing**?"

Reply:
...and the whole point of `Ty::Error` is to stop more errors from being emitted. That was some great code from me lol.